tests/phpunit/tests/post/wpUpdatePost.php

```diff
@@ -0,0 +1,237 @@
+	 * and is different from the existing tags.
+	 *
+	 * @ticket 45121
+	 */
+	public function test_update_post_should_only_modify_post_tags_if_different_tags_input_was_provided() {
+		$tag_1 = wp_insert_term( 'wp_update_post_tag', 'post_tag', array( 'slug' => 'wp_update_post_tag_1' ) );
+		$tag_2 = wp_insert_term( 'wp_update_post_tag', 'post_tag', array( 'slug' => 'wp_update_post_tag_2' ) );
+		$tag_3 = wp_insert_term( 'wp_update_post_tag', 'post_tag', array( 'slug' => 'wp_update_post_tag_3' ) );
+
+		$post_id = self::factory()->post->create(
+			array(
+				'tags_input' => array( $tag_1['term_id'], $tag_2['term_id'] ),
+			)
+		);
+
+		$post = get_post( $post_id );
+
+		$tags = wp_get_post_tags( $post->ID, array( 'fields' => 'ids' ) );
+		$this->assertSameSets( array( $tag_1['term_id'], $tag_2['term_id'] ), $tags );
+
+		wp_update_post( $post );
+
+		$tags = wp_get_post_tags( $post->ID, array( 'fields' => 'ids' ) );
+		$this->assertSameSets( array( $tag_1['term_id'], $tag_2['term_id'] ), $tags );
+
+		wp_update_post(
+			array(
+				'ID'         => $post->ID,
+				'tags_input' => array( $tag_2['term_id'], $tag_3['term_id'] ),
+			)
+		);
+
+		$tags = wp_get_post_tags( $post->ID, array( 'fields' => 'ids' ) );
+		$this->assertSameSets( array( $tag_2['term_id'], $tag_3['term_id'] ), $tags );
+	}
+
+	/**
+	 * Test the wp_update_post() filters post content when 'post_status' is 'draft'.
+	 *
+	 * @ticket 22944
+	 */
+	public function test_wp_update_post_with_content_filtering() {
+		kses_remove_filters();
+
+		$post_id = wp_insert_post(
+			array(
+				'post_title' => '<script>Test</script>',
+			)
+		);
+		$post    = get_post( $post_id );
+		$this->assertSame( '<script>Test</script>', $post->post_title );
+		$this->assertSame( 'draft', $post->post_status );
+
+		kses_init_filters();
+
+		wp_update_post(
+			array(
+				'ID'          => $post->ID,
+				'post_status' => 'publish',
+			)
+		);
+
+		kses_remove_filters();
+
+		$post = get_post( $post->ID );
+		$this->assertSame( 'Test', $post->post_title );
+	}
+
+	/**
+	 * Test updating a post and preserving post_date when changing to future status
+	 *
+	 * @ticket 62468
+	 *
+	 * @dataProvider data_post_status_transitions
+	 *
+	 * @param string $initial_status Initial post status.
+	 */
+	public function test_update_post_preserves_date( $initial_status ) {
+
+		$post_id = self::factory()->post->create(
+			array(
+				'post_status' => $initial_status,
+			)
+		);
+
+		$future_date = gmdate( 'Y-m-d H:i:s', strtotime( '+1 day' ) );
+		$update_data = array(
+			'ID'          => $post_id,
+			'post_status' => 'future',
+			'post_date'   => $future_date,
+		);
+
+		wp_update_post( $update_data );
+		$updated_post = get_post( $post_id );
+
+		$this->assertSame( $future_date, $updated_post->post_date );
+	}
+
+	/**
+	 * Data provider for test_update_post_preserves_date
+	 *
+	 * @return array[] Test parameters
+	 */
+	public function data_post_status_transitions() {
```

Comment:
This ought to have the same name as the test with the `test_` prefix removed and replaced with `data_`.